lib/text/vtt_text_parser.js

```diff
@@ -843,6 +844,37 @@ shaka.text.VttTextParser = class {
 
     return (milliseconds / 1000) + seconds + (minutes * 60) + (hours * 3600);
   }
+
+  /**
+   * This method converts the HTML entities &amp;, &lt;, &gt;, &quot;, and
+   * &#39; in string to their corresponding characters.
+   *
+   * @param {!string} input
+   * @return {string}
+   * @private
+   */
+  static htmlUnescape_(input) {
+    // Used to map HTML entities to characters.
+    const htmlUnescapes = {
+      '&amp;': '&',
+      '&lt;': '<',
+      '&gt;': '>',
+      '&quot;': '"',
+      '&#39;': '\'',
+    };
+
+    // Used to match HTML entities and HTML characters.
+    const reEscapedHtml = /&(?:amp|lt|gt|quot|#(0+)?39);/g;
```

Comment:
I think `#(0+)?39;` can match much more than the dictionary entry `#39;` above.  Is that why you're defaulting to `\'` below if the dictionary lookup fails?

Reply:
yes, but if you have a better idea I accept it, you are more expert in regular expressions

Reply:
@joeyparrish :)

Reply:
I'll see if I can come up with a suggestion.  If not, I'll add a comment and approve it.

Reply:
None of the things I thought of seem worth it.  I'll add a comment below and call it good enough.